crates/net/eth-wire-types/src/status.rs

```diff
@@ -306,6 +306,95 @@ impl From<Status> for StatusEth69 {
     }
 }
 
+impl From<StatusEth69> for Status {
+    fn from(s: StatusEth69) -> Self {
+        Self {
+            version: s.version,
+            chain: s.chain,
+            // Eth69 doesn't include total_difficulty so we choose a default
+            total_difficulty: U256::default(),
+            blockhash: s.blockhash,
+            genesis: s.genesis,
+            forkid: s.forkid,
+        }
+    }
+}
+
+/// `StatusMessage` can store either the Legacy version (with TD) or the
+/// eth/69 version (omits TD).
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum StatusMessage {
+    /// The legacy status (`eth/66` through `eth/68`) with `total_difficulty`.
+    Legacy(Status),
+    /// The new `eth/69` status with no `total_difficulty`.
+    Eth69(StatusEth69),
+}
+
+impl StatusMessage {
+    /// Returns the genesis hash from the status message.
+    #[inline]
+    pub fn genesis(&self) -> FixedBytes<32> {
+        match self {
+            Self::Legacy(legacy_status) => legacy_status.genesis,
+            Self::Eth69(status_69) => status_69.genesis,
+        }
+    }
+
+    /// Returns the protocol version.
+    #[inline]
+    pub fn version(&self) -> EthVersion {
+        match self {
+            Self::Legacy(legacy_status) => legacy_status.version,
+            Self::Eth69(status_69) => status_69.version,
+        }
+    }
+
+    /// Returns the chain identifier.
+    #[inline]
+    pub fn chain(&self) -> &Chain {
+        match self {
+            Self::Legacy(legacy_status) => &legacy_status.chain,
+            Self::Eth69(status_69) => &status_69.chain,
+        }
+    }
+
+    /// Returns the fork identifier.
+    #[inline]
+    pub fn forkid(&self) -> ForkId {
+        match self {
+            Self::Legacy(legacy_status) => legacy_status.forkid,
+            Self::Eth69(status_69) => status_69.forkid,
+        }
+    }
+
+    /// Converts to legacy Status since full support for EIP-7642
+    /// is not fully implemented
+    /// `<https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7642.md>`
+    pub fn to_legacy(self) -> Status {
+        match self {
+            Self::Legacy(s) => s,
+            Self::Eth69(s) => s.into(),
```

Comment:
I'd like to move the from impl here and remove the from impl, this way it isn't as obfuscated that this isn't exactly right

Reply:
I feel you on that. ported over the conversion to legacy status from the impl to here: https://github.com/paradigmxyz/reth/blob/339dca1188514e7a250add41f1ae6db965c20239/crates/net/eth-wire-types/src/status.rs#L360-L373